core/src/main/java/nl/nn/adapterframework/extensions/kafka/KafkaListener.java

```diff
@@ -0,0 +1,145 @@
+		properties.setProperty(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, fromBeginning ? "earliest" : "latest");
+		properties.setProperty(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class.getName());
+		properties.setProperty(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, ByteArrayDeserializer.class.getName());
+		properties.setProperty(ConsumerConfig.METADATA_MAX_AGE_CONFIG, String.valueOf(patternRecheckInterval));
+		consumer = new KafkaConsumer<>(properties);
+	}
+	private void onMessage(ConsumerRecord<String, byte[]> record) {
+		try (PipeLineSession session = new PipeLineSession()) {
+			messageHandler.processRawMessage(this, wrapRawMessage(record, session), session, false);
+		} catch (Exception e) {
+			ibisExceptionListener.exceptionThrown(this, e);
+		}
+	}
+
+	@Override
+	public void open() throws ListenerException {
+		Arrays.stream(topics.split(","))
+				.filter(StringUtils::isNotEmpty) //if topics is `abc` java will split it into `abc` and ``. The latter is not a valid topic.
+				.map(Pattern::compile) //Convert the topic to a regex pattern, to allow for wildcards in topic names.
+				.forEach(consumer::subscribe);
+		poller = new KafkaInternalConsumer(consumer, pollTimeout, this::onMessage);
+	}
+
+	@Override
+	public void close() throws ListenerException {
+		poller.kill();
+	}
+
+	@Override
+	public Message extractMessage(
+			@Nonnull RawMessageWrapper<ConsumerRecord<String, byte[]>> rawMessage, @Nonnull Map<String, Object> context) throws ListenerException {
+		return new Message(rawMessage.getRawMessage().value());
+	}
+
+	@Override
+	public void afterMessageProcessed(PipeLineResult processResult, RawMessageWrapper<ConsumerRecord<String, byte[]>> rawMessage, PipeLineSession pipeLineSession) throws ListenerException {
+		// no-op
+	}
+
+	@Override
+	public void setHandler(IMessageHandler<ConsumerRecord<String, byte[]>> handler) {
+		this.messageHandler = handler;
+	}
+
+	@Override
+	public void setExceptionListener(IbisExceptionListener listener) {
+		this.ibisExceptionListener = listener;
+	}
+
+	@Override
+	public RawMessageWrapper<ConsumerRecord<String, byte[]>> wrapRawMessage(ConsumerRecord<String, byte[]> rawMessage, PipeLineSession session) throws ListenerException {
+		Map<String, String> headers=new HashMap<>();
+		Arrays.stream(rawMessage.headers().toArray()).forEach(header -> {
+			try {
+				headers.put(header.key(), new String(header.value(), StandardCharsets.UTF_8));
+			} catch(Exception e) {
+				log.warn("Failed to convert header key [{}] to string. Bytearray value: [{}]", header.key(), header.value(), e);
+			}
+		});
+		Map<String, Object> context = new HashMap<>();
+		context.put("kafkaTopic", rawMessage.topic());
+		context.put("kafkaKey", rawMessage.key());
+		context.put("kafkaPartition", rawMessage.partition());
+		context.put("kafkaOffset", rawMessage.offset());
+		context.put("kafkaTimestamp", rawMessage.timestamp());
+		context.put("kafkaHeaders", headers);
```

Comment:
Heb je deze nodig? Wat is de reden dat dit bewaard wordt? Het voelt meer als data die je in de MessageContext zou willen zien. Dit zou echt het minimale moeten zijn dat je nodig hebt. Het uitlezen van het daadwerkelijke bericht gebeurd pas in `extractMessage`.

Reply:
Aangepast, staat nu in extractMessage.